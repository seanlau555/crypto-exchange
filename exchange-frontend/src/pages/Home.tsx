import { Text, Button, Flex, Box, Heading, Container } from '@chakra-ui/react'
import { ButtonEvent } from '../types'

export const alpacaAuthUrl = 'https://app.alpaca.markets/oauth/authorize'

function Home() {
  const handleSubmit = async (evt: ButtonEvent) => {
    evt.preventDefault()
    const alpaca_oauth =
      alpacaAuthUrl +
      `?response_type=code&client_id=${
        import.meta.env.VITE_REACT_APP_CLIENT_ID
      }&redirect_uri=${
        import.meta.env.VITE_REACT_APP_REDIRECT_URI
      }&scope=account:write%20trading%20data`
    document.location.href = alpaca_oauth
  }

  return (
    <Container maxW="6xl">
      <Flex
        align="flex-start"
        justify="center"
        direction="column"
        mt={{ base: 8 }}
      >
        <Heading> My OAuth Trading App</Heading>
        <Box mt="10px">
          <Button
            data-testid="signin"
            colorScheme="teal"
            onClick={handleSubmit}
          >
            Sign in with Alpaca
          </Button>
        </Box>
        <Box mt="10px">
          <Text fontWeight="bold"> Powered by Alpaca </Text>
        </Box>
      </Flex>
    </Container>
  )
}

export default Home
