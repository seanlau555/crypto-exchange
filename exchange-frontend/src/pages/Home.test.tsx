import { describe, expect, it } from 'vitest'
import Home from './Home'
import { screen, render, fireEvent } from '@testing-library/react'

describe('<Home />', () => {
  it('render home page', () => {
    const { container } = render(<Home />)
    expect(container).toHaveTextContent('My OAuth Trading App')
  })

  it('should able to click submit button', () => {
    render(<Home />)
    const buttonElement = screen.getByRole('button', {
      name: /Sign in with Alpaca/i,
    })
    fireEvent.click(buttonElement)
    expect(window.location.href).toEqual('http://localhost:3000/')
  })
})
