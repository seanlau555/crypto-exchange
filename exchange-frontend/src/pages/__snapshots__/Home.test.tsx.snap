// Vitest Snapshot v1

exports[`<Home /> > render home page 1`] = `
<div>
  <div
    class="chakra-container css-12irvte"
  >
    <div
      class="css-1d7sxvi"
    >
      <h2
        class="chakra-heading css-0"
      >
         My OAuth Trading App
      </h2>
      <div
        class="css-1q7njkh"
      >
        <button
          class="chakra-button css-5c2176"
          data-testid="signin"
          type="button"
        >
          Sign in with Alpaca
        </button>
      </div>
      <div
        class="css-1q7njkh"
      >
        <p
          class="chakra-text css-in3yi3"
        >
           Powered by Alpaca 
        </p>
      </div>
    </div>
  </div>
</div>
`;
